geoalchemy2/elements.py

```diff
@@ -131,11 +132,35 @@ class WKBElement(_SpatialElement):
 
     Note: you can create ``WKBElement`` objects from Shapely geometries
     using the :func:`geoalchemy2.shape.from_shape` function.
+
+
+    WKB byte_order:
+        - WKB_XDR = 0  # Most Significant Byte First
+        - WKB_NDR = 1  # Least Significant Byte First
+
     """
 
     geom_from = 'ST_GeomFromWKB'
     geom_from_extended_version = 'ST_GeomFromEWKB'
 
+    def __init__(self, data, srid=-1, extended=False):
+        if extended and srid == -1:
+            # unpack srid from WKB struct
+
+            try:
+                # support hex input
+                data = binascii.unhexlify(data[:18])
```

Comment:
@sbouchut I just realized that we should create a specific variable instead of erasing `data` because it is then passed to `_SpatialElement.__init__()` so in the end the WKBElement.data will be ill-formed. Note that the tests do not point this unexpected behaviour, maybe we should add a test for that.

Reply:
agree, thanks for the feedback